File: 371. Sum of Two Integers/solution.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var getSum = function(a, b) {
    while (b !== 0) {
        var x = a & b;
        a = a ^ b;
        b = x << 1;
    }
    return a;
};

//console.log(getSum(1, 2));

File: 371. Sum of Two Integers/solution.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        if a <= 0 or b <= 0:
            return a + b
        carry = a & b
        result = a ^ b
        while carry:
            carry_t = carry << 1
            carry = carry_t & result
            result ^= carry_t
        return result
